base/Dockerfile

```diff
@@ -2,11 +2,34 @@
 FROM gcr.io/google-appengine/debian8
 
 # Install updates and dependencies
-RUN apt-get update -y && apt-get install --no-install-recommends -y -q curl python build-essential git ca-certificates libkrb5-dev imagemagick netbase && \
-    apt-get clean && rm /var/lib/apt/lists/*_*
+RUN apt-get update -y && \
+    apt-get install --no-install-recommends -y -q \
+      build-essential \
+      ca-certificates \
+      curl \
+      git \
+      imagemagick \
+      libkrb5-dev \
+      netbase \
+      python && \
+    apt-get clean && \
+    rm /var/lib/apt/lists/*_*
+
+# Add the files necessary for verifying and installing node
+# These files are placed in /opt/google_appengine/runtime so that
+# they will remain even if the /nodejs directory is removed.
+# Further, the PATH is is not updated, because these scripts should
+# only be called using an absolute path.
+ADD contents/ /opt/google_appengine/runtime/
```

Comment:
nit: it would be better to call this `gcp/runtime` or `google_cloud/runtime` instead of `google_appengine` because the image can be used in other places too.